tests/test_pmem_allocator.cpp

```diff
@@ -0,0 +1,201 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2021 Intel Corporation
+ */
+
+#include <chrono>
+#include <ctime>
+#include <deque>
+#include <future>
+#include <string>
+#include <sys/time.h>
+#include <thread>
+#include <vector>
+
+#include "../engine/kv_engine.hpp"
+#include "../engine/logger.hpp"
+#include "../engine/thread_manager.hpp"
+
+#include "gtest/gtest.h"
+
+#include "kvdk/engine.hpp"
+#include "pmem_allocator/free_list.hpp"
+#include "pmem_allocator/pmem_allocator.hpp"
+#include "test_util.h"
+
+using namespace KVDK_NAMESPACE;
+
+class EnginePMemAllocatorTest : public testing::Test {
+protected:
+  Engine *engine = nullptr;
+  Configs configs;
+  std::shared_ptr<ThreadManager> thread_manager_;
+  std::string pmem_path;
+
+  virtual void SetUp() override {
+    pmem_path = "/mnt/pmem0/kvdk_pmem_allocator";
+    GlobalLogger.Init(stdout, LogLevel::All);
+    char cmd[1024];
+    sprintf(cmd, "rm -rf %s\n", pmem_path.c_str());
+    int res __attribute__((unused)) = system(cmd);
+  }
+
+  virtual void TearDown() { // delete db_path
+    char cmd[1024];
+    sprintf(cmd, "rm -rf %s\n", pmem_path.c_str());
+    int res __attribute__((unused)) = system(cmd);
+  }
+};
+
+TEST_F(EnginePMemAllocatorTest, TestBasicAlloc) {
+  uint64_t pmem_size = 128ULL << 20; // 128MB
+  uint64_t alloc_size = 8;
+
+  // params config
+  std::vector<uint64_t> num_segment_blocks{1024, 2 * 1024, 2 * 1024 * 1024};
+  std::vector<uint32_t> block_sizes{16, 32, 64};
+  std::vector<uint32_t> num_threads = {1, 16};
+
+  for (int i = 0; i < num_segment_blocks.size(); ++i) {
+    for (auto num_thread : num_threads) {
+      thread_manager_.reset(new (std::nothrow) ThreadManager(num_thread));
```

Comment:
Check nullptr for nothrow, or just let allocator do its job -- throw std::bad_alloc

Reply:
remove std::nothrow